src/vs/workbench/contrib/tasks/browser/abstractTaskService.ts

```diff
@@ -324,15 +324,10 @@ export abstract class AbstractTaskService extends Disposable implements ITaskSer
 			}
 			return task._label;
 		});
-		this._lifecycleService.onBeforeShutdown(e => {
+		this._lifecycleService.onBeforeShutdown(async e => {
 			this._willRestart = e.reason !== ShutdownReason.RELOAD;
-			this.getActiveTasks().then(async (tasks) => {
-				for (const task of tasks) {
-					if (!task.configurationProperties.isBackground) {
-						await this.terminate(task);
-					}
-				}
-			});
+			const tasks = (await this.getActiveTasks()).filter(t => t.configurationProperties.isBackground);
+			await Promise.all(tasks.map(t => this.terminate(t)));
```

Comment:
Just for my education...

I get why you don't want to `await this.terminate(task)` each time through the loop. But what's the point of the `await Promise.all` in this case? Nothing (as far as I can see) is going to await the result of the callback, so why not just

```
  const tasks = (await this.getActiveTasks()).filter(t => t.configurationProperties.isBackground);
  for (const task of tasks) {
    this.terminate(task);
  }
``` 
If I'm wrong, and something *is* going to await it, why not `return Promise.all(...)` rather than `await Promise.all(...)`